modules/statistics-service-api/src/main/java/org/opencastproject/statistics/api/StatisticsProviderRegistry.java

```diff
@@ -39,4 +39,17 @@
    * @param provider The provider to remove.
    */
   void removeProvider(StatisticsProvider provider);
+
+  /**
+   * Add a new writer instance
```

Comment:
I don't find it intuitive that the ServiceProviderRegistry also handles the writers. Maybe move all of this into the general StatisticsService interface instead, or is there a need for a specific registry interface?

Reply:
I admit, the relationships between the statistics interfaces are a little convoluted. The thing about `StatisticsService` (and it’s raison d’etre) is, it has a remote implementation to retrieve statistics. This makes no sense for the `StatisticsProviderRegistry`, because how would you add a provider, which is a piece of Java code, remotely? So, if we were to merge `StatisticsService` and the provider registry, we’d have to throw an exception for the methods we’re not implementing, which isn’t intuitive as well.

Granted, we could add *another* interface `StatisticsWriterRegistry` and let `StatisticsServiceImpl` implement that, but I don’t see that much need for it, given that the current registry isn’t overflowing with methods to implement.

Reply:
Ah, I see now. In that case, maybe we can come up with a more general name than StatisticsProviderRegistry? Although I have to admit that I can't think of one right now...

Reply:
StatisticsWriterRegistry is the only thing I can think of, and that's not any better :D

Reply:
@gregorydlogan Well, it's not just for writers.

Anybody against `StatisticsCoordinator`? It doesn't feel wise to hold up this PR because of a naming issue. At least it's not a `StatisticsProviderFactorySingletonInstanceObserver` :D

Reply:
Fine with me.

Reply:
Sounds good to me.